security/pkg/server/ca/server.go

```diff
@@ -164,3 +175,21 @@ func Authenticate(ctx context.Context, auth []security.Authenticator) *security.
 	serverCaLog.Warnf("Authentication failed for %v: %s", getConnectionAddress(ctx), errMsg)
 	return nil
 }
+
+// find the root cert from cert chain
+func findRootCertFromCertificateChainBytes(certBytes []byte) ([]byte, error) {
+	var cert []byte
+	var block *pem.Block
+	for {
+		block, certBytes = pem.Decode(certBytes)
+		if len(certBytes) == 0 {
+			break
+		}
+		_, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
```

Comment:
This isn't sufficient IMO. We need to add checks here to ensure that the cert returned from this function is a ca.crt

Reply:
verify cert now